nvme.c

```diff
@@ -36,6 +35,7 @@
 #include <unistd.h>
 #include <math.h>
 #include <dirent.h>
+#include <libgen.h>
```

Comment:
Keep in mind that on Linux/glibc, `<libgen.h>` activates the POSIX version of basename (that mutates the string), while it seems to have the exact opposite effect on FreeBSD 12. (Just today was dealing with this in libinput :D)

So it might be a good idea to `strdup` the argument of `basename` below

Reply:
In both it activates the non-const version of basename (I couldn't actually find a const version in Linux, but I may have looked in the wrong places). Both will modify the argument, but if you trace back where the argument actually came from, and when such modifications take place, you'll see there's nothing to worry about. It took me a bit of tracing through the code to convince myself that this is correct in both places, both in terms of implementation, and in terms of what the spec says.